assets/src/components/resource/editors/OutlineItem.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { PropsWithChildren, useRef } from 'react';
```

Comment:
Common outline item components and functions were broken out into this file and are imported into the different editor files as needed